test/sync-controller.test.js

```diff
@@ -246,30 +246,20 @@ QUnit.test('ProgramDateTime strategy finds nearest llhls sync point', function(a
 
   this.syncController.setDateTimeMappingForStart(playlist);
 
-  const newPlaylist = playlistWithDuration(200, {llhls: true});
-
-  syncPoint = strategy.run(this.syncController, newPlaylist, duration, timeline);
-
-  assert.equal(syncPoint, null, 'no syncpoint when datetimeObject not set on playlist');
-
-  newPlaylist.segments.forEach((segment, index) => {
-    segment.dateTimeObject = new Date(2012, 11, 12, 12, 12, 22 + (index * 10));
-  });
-
-  syncPoint = strategy.run(this.syncController, newPlaylist, duration, timeline, 194);
+  syncPoint = strategy.run(this.syncController, playlist, duration, timeline, 194);
 
   assert.deepEqual(syncPoint, {
     time: 192,
-    segmentIndex: 18,
+    segmentIndex: 19,
```

Comment:
Do you know why this and below changed even though the playlists look like they should be the same?

Reply:
This happened because newPlaylist had a `22 + (index * 10)` instead of a `12 + (index * 10)` for its `dateTimeObject`. I think that using a weird date like that makes it much harder to understand what the test is doing. The way it's setup now its easy to see why the sync points are where they are.